charts/kong/README.md

```diff
@@ -343,13 +466,17 @@ individual services: see values.yaml for their individual default values.
 * `manager`
 * `portal`
 * `portalapi`
+* `cluster`
 * `status`
 
 `status` is intended for internal use within the cluster. Unlike other
 services it cannot be exposed externally, and cannot create a Kubernetes
 service or ingress. It supports the settings under `SVC.http` and `SVC.tls`
 only.
 
+`cluster` is used on hybrid mode control plane nodes. It does not support the
+`SVC.http.*` settings, as cluster communications must be TLS-only.
+
```

Comment:
suggestion: explain what kind of data the `cluster` service is used for. Imagine that a security engineer is reading this and is trying to tell whether the business (data plane) data may flow through this channel.

Reply:
Noted, though outside our area of concern--if the protocol details and contents of data transferred do come up in customer questions, it's something that should live on the main doc site. There's some overlap between content there and here, but content here should skew more towards chart-specific stuff. This is mostly to avoid doc rot: the main doc site gets more thoroughly reviewed during Kong releases and is better able to ensure it keeps up to date with changes as they arrive. That's harder for us since we don't track the Kong release versions: chart releases happen on their own schedule and apply to multiple Kong releases and variants, so the docs here are lighter on the specifics.

Reply:
Maybe--it's something I would defer absent evidence it's needed from the field. If we start getting a lot of questions about the details of the protocol, it's something we should add; if it doesn't come up often, then it probably isn't needed. That information may be of interest to a specialized audience, but someone just trying to get hybrid mode running shouldn't need to know detail beyond it using TLS and requiring TLS configuration.

Brevity is desirable, and I'll avoid including finer details if I think it's likely most readers would likely gloss over them anyway. My personal experience from the field (and in the comments here :smile:) is that complaints are more often directed towards a lack of simple instructions, where my response is "the simple instructions _are_ there, but they're alongside a lot of other detail, and I can understand why it's easy to miss them when looking at the large document". Because of that, if I think I can reasonably omit something and still guide readers to my goal (here, that they can spin up a hybrid mode cluster), I will.

Reply:
> the simple instructions are there, but they're alongside a lot of other detail, and I can understand why it's easy to miss them when looking at the large document

Imagine a case where a Kubernetes user (new to Kong) comes to this repo to try out Kong in hybrid mode.
- Is it a likely scenario?
- Where are the "simple instructions" for this case? Most of my clarification requests originate from the fact that I'm consistently failing to locate those simple instructions, per my definition of simplicity (which can be roughly summarized by "contained or, at least, listed with links in one place")

Maybe I'm bikeshedding or missing the scope here and it's appropriate to curb this discussion. But my (newcomer's) perception is pretty clear that the documentation does not help to get up and running easily (except for a trivial case of "everything by default", but even in that case it doesn't apprently give any hint about how to make that "default" Kong do anything useful, like proxy to a vanilla echoserver present in the cluster).

In other words, one of the frameworks classifies documentation into four categories: how-tos, tutorials, references and explanations. This document serves as a good reference. I miss how-tos and tutorials for Kong Helm charts.

Reply:
> likely

Ideally no. The core product here is Kong, rather than the chart, and we sort of want users to come in through konghq.com and docs.konghq.com, since that's where all the flashy marketing content and basic introductions live.

The README.md here has some of that, because people do use it as a starting point in practice, but there are challenges to making it an effective entrypoint because there are so many branches: when you throw in "ah, but what if Enterprise?", "ah, but what if the controller is on?", "ah, but what if you use an HTTP load balancer versus a TCP load balancer?", etc.. There are a _lot_ of complicating factors that can and do come up, so when asked, "could we provide simpler instructions?" my response is often "yes, if you want to reduce the product feature set and allowed feature combinations" or "yes, if you want to focus on a single set of features in a fairly prescribed environment, which isn't the purpose of this document".

README.md is a general overview that seeks to cover many of these at a high level, so it can't also be specific. That's where (future) guides would come in, since they can more easily state up front that the guide specifically covers an Enterprise hybrid configuration without a controller, and then instructs users to create a TCP load balancer for the proxy.

Recognizing that it does have a dual purpose, there are simple instructions, but they're very basic, and specifically avoid those branching paths. That's why the [install instructions](https://github.com/Kong/charts/tree/master/charts/kong#install) and [Enterprise instructions](https://github.com/Kong/charts/tree/master/charts/kong#kong-enterprise) comprise only must-do, your-cluster-will-not-start-at-all-regardless-of-other-configuration-if-you-ignore-this steps, to present those absolute requirements with as little extra info as possible, since we want users (and particularly less savvy users) to be able to get to solid baseline before proceeding.